test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -55,6 +55,49 @@ public async Task SingleOperationTriggerTest()
             changes.Clear();
         }
 
+        /// <summary>
+        /// Verifies that manually setting the batch size correctly changes the number of changes processed at once
+        /// </summary>
+        [Fact]
+        public async Task BatchSizeOverrideTriggerTest()
+        {
+            this.EnableChangeTrackingForTable("Products");
+            this.StartFunctionHost(nameof(ProductsTriggerWithValidation), Common.SupportedLanguages.CSharp, true, environmentVariables: new Dictionary<string, string>() {
+                { "TEST_EXPECTED_BATCH_SIZE", "20" },
+                { "Sql_Trigger_BatchSize", "20" }
+            });
+
+            var changes = new List<SqlChange<Product>>();
+            this.MonitorProductChanges(changes, "SQL Changes: ");
+
+            // Considering the polling interval of 5 seconds and batch-size of 20, it should take around 10 seconds to
+            // process 40 insert operations.
+            this.InsertProducts(1, 40);
+            await Task.Delay(TimeSpan.FromSeconds(12));
```

Comment:
I plan on coming back and refactoring this logic to not be a hardcoded wait since that's more unstable and less reliable. Will be done separately though since I want to update all the places that are using this kind of logic at once though.

Reply:
If you are planning on updating the wait-logic, should I keep the changes in my PR ready to be committed? I have been busy with Hackathon so haven't picked up any work on SQL extension items this week.